src/btree/bt_sync.c

```diff
@@ -248,6 +250,29 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
                     btree->rec_max_txn = mod->rec_max_txn;
                 if (mod != NULL && btree->rec_max_timestamp < mod->rec_max_timestamp)
                     btree->rec_max_timestamp = mod->rec_max_timestamp;
+
+                /*
+                 * Check whether the tree is lookaside btree and verify whether the page contents
+                 * are obsolete or not by checking the visibility of the page stop time pair.
+                 */
+                addr = walk->addr;
+                if (is_las &&
+                  __wt_txn_visible_all(session, addr->newest_stop_txn, addr->newest_stop_ts)) {
+                    /* Initialize the modify structure if it doesn't exist */
+                    if (mod == NULL)
+                        WT_ERR(__wt_page_modify_init(session, walk->page));
+
+                    /*
+                     * Mark the page as empty and also set the parent page as dirty. This is to
+                     * ensure when the parent page is checkpointing, the empty child page will be
+                     * cleaned.
+                     */
+                    if (!WT_PAGE_IS_INTERNAL(walk->page)) {
+                        walk->page->modify->rec_result = WT_PM_REC_EMPTY;
```

Comment:
Is setting the page as empty directly without reconciling code is fine?

Reply:
@keithbostic could you take a look at Hari's question?

Reply:
I don't think it's dangerous, but I don't think it's a useful optimization to do in the checkpoint code.

Why not have reconciliation do the same check? (We'd want to fast-track LAS file eviction too, I think.)

So, I wouldn't try and fast-path it here.

Reply:
@keithbostic, Sorry I didn't understand your point clearly. Currently, only the clean pages are checked for obsolete in the checkpoint as they are skipped to pass to reconciliation. So moving this part of the code to reconciliation by changing to ensure all the pages go to reconciliation even if it is clean and handle this obsolete check there?

Reply:
@kommiharibabu, of course -- thank you for pointing that out!

I would still not do this work here. It's not saving us any work in this checkpoint, and the "normal" work of handling empty pages wouldn't get done (for example, reverse splits up the tree).

Please let me know if I'm missing the idea, that there is a win here by getting the checkpoint done more quickly, or more simply.

Maybe it would be reasonable for eviction to take into account the visibility of the LAS pages when considering the pages to choose for eviction? And, of course, reconciliation could fast-track and not do any work.

Reply:
@kommiharibabu, I hadn't read WRITING-3405, so I didn't understand what you're trying to do here.

I think I understand better now.

I think the way this has to be work is by setting `WT_REF.state = WT_REF_DELETED` for the leaf page. Ideally, we can use the `WT_REF_DELETED` state and not invent a new one. The tradeoff is that much of the reconciliation code is already written to handle the `WT_REF_DELETED` state, but obviously there may not be enough overlap between fast truncate and discarding history file pages for this to work.

It seems to me that checkpoint (`bt_sync.c`) is going to switch the leaf page `WT_REF.state` field (which may require locking and/or atomic operations), then reconciliation is going to handle the rest. Specifically, `__rec_child_modify()` on the internal parent page is going to notice the `WT_REF.state` (https://github.com/wiredtiger/wiredtiger/blob/develop/src/reconcile/rec_child.c#L141-L154), call `__rec_child_deleted()`, which is going to actually free the underlying block in the checkpoint (https://github.com/wiredtiger/wiredtiger/blob/develop/src/reconcile/rec_child.c#L72).

We could duplicate the code in reconciliation to handle a new state, but I think that's going to be more work overall than teasing apart `WT_REF.state = WT_REF_DELETED` with and without an associated `WT_REF.page_del` value.

Reply:
@keithbostic, Thanks for your input. I implemented according to your suggestion for a leaf page that has `WT_ADDR` currently. Created an another ticket to handle the scenario of the page is in-memory and has multiple blocks that are obsolete.